plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-products-catalog-controller.php

```
@@ -0,0 +1,273 @@
+		return $this->catalog_generation_response( $file_info );
+	}
+
+	/**
+	 * Checks if a given request has permission to request products catalog.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|bool
+	 *
+	 * @internal For exclusive usage within this class, backwards compatibility not guaranteed.
+	 */
+	public function request_catalog_permissions_check( $request ) {
```

PHPMD (2.15.0)

Avoid unused parameters such as '$request'. (undefined)

(UnusedFormalParameter)